stackslib/src/chainstate/nakamoto/miner.rs

```diff
@@ -330,74 +227,36 @@ impl NakamotoBlockBuilder {
-                    ),
-                )?
-                .ok_or(Error::NoSuchBlockError)
-                .map_err(|e| {
-                    warn!(
-                        "No such Nakamoto parent block {}/{} ({})",
-                        &nakamoto_parent_header.consensus_hash,
-                        &nakamoto_parent_header.block_hash(),
-                        &nakamoto_parent_header.block_id()
-                    );
-                    e
-                })?;
-
-                (
-                    parent_header_info,
-                    nakamoto_parent_header.consensus_hash.clone(),
-                    nakamoto_parent_header.block_hash(),
-                )
-            } else if let Some((stacks_header, consensus_hash)) = self.epoch2_parent_header.as_ref()
-            {
-                // parent is a Stacks epoch2 block
-                let parent_header_info = NakamotoChainState::get_block_header(
-                    chainstate.db(),
-                    &StacksBlockId::new(consensus_hash, &stacks_header.block_hash()),
-                )?
-                .ok_or(Error::NoSuchBlockError)
-                .map_err(|e| {
-                    warn!(
-                        "No such Stacks 2.x parent block {}/{} ({})",
-                        &consensus_hash,
-                        &stacks_header.block_hash(),
-                        &StacksBlockId::new(&consensus_hash, &stacks_header.block_hash())
-                    );
-                    e
-                })?;
-
-                (
-                    parent_header_info,
-                    consensus_hash.clone(),
-                    stacks_header.block_hash(),
-                )
-            } else {
+        let (chain_tip, parent_consensus_hash, parent_header_hash) = match self.parent_header {
+            Some(ref header_info) => (
+                header_info.clone(),
+                header_info.consensus_hash.clone(),
+                header_info.anchored_header.block_hash(),
+            ),
+            None => {
                 // parent is genesis (testing only)
                 (
                     StacksHeaderInfo::regtest_genesis(),
                     FIRST_BURNCHAIN_CONSENSUS_HASH.clone(),
                     FIRST_STACKS_BLOCK_HASH.clone(),
                 )
-            };
+            }
+        };
 
-        let coinbase_height = if let Ok(Some(parent_coinbase_height)) =
-            NakamotoChainState::get_coinbase_height(
-                chainstate.db(),
-                &StacksBlockId::new(&parent_tenure_id_consensus_hash, &parent_header_hash),
-            ) {
+        let parent_block_id = StacksBlockId::new(&parent_consensus_hash, &parent_header_hash);
+        let parent_coinbase_height =
+            NakamotoChainState::get_coinbase_height(chainstate.db(), &parent_block_id)
+                .ok()
+                .flatten()
+                .unwrap_or(0);
+
+        let new_tenure = cause == Some(TenureChangeCause::BlockFound);
```

Comment:
This is a total nitpick, but is there a reason to assign cause == Some(Tenure..) to a var?

Reply:
I think it makes the if statement below easier to read. Probably just a personal preference for me, but I find binary comparisons inside of rust's ternary-like structures kind of hard to read:

```rust
let coinbase_height = if cause == Some(TenureChangeCause::BlockFound) {
   ...
}
```

Reply:
I usually just rely on comments in this instance but it was more to make sure I wasn't missing something. So sounds good :D